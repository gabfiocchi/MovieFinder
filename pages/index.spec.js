/* global jest describe, it, expect */
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import HomePage from './index.vue'

const mockRouterPush = jest.fn()
const localVue = createLocalVue()
localVue.use(Vuex)

const factory = () => {
  const store = new Vuex.Store({
    modules: {
      movies: {
        state: {
          isLoading: false
        }
      }
    }
  })
  return shallowMount(HomePage, {
    localVue,
    store,
    stubs: {
      fa: true
    },
    mocks: {
      $router: {
        push: mockRouterPush
      }
    }
  })
}

describe('Home Page', () => {
  it('should render the component', () => {
    // Arrange.
    // Act.
    const wrapper = factory()
    // Assert.
    expect(wrapper).toMatchSnapshot()
  })

  it('should be initialize without query', () => {
    // Arrange.
    // Act.
    const wrapper = factory()
    // Assert.
    expect(wrapper.vm.query).toBe('')
  })

  it('searchMovie', async () => {
    // Arrange.
    // Act.
    const wrapper = factory()
    await wrapper.find('input').setValue('Star Wars')
    wrapper.find('form').trigger('submit.prevent')
    // Assert.
    expect(wrapper.vm.query).toBe('Star Wars')
    expect(mockRouterPush).toHaveBeenCalled()
  })
})
