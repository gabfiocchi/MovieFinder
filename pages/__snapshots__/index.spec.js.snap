// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render the component 1`] = `
<div class="container">
  <div class="w-full text-center">
    <h1 class="subtitle">
      Movie Finder
    </h1>
    <!---->
    <form class="max-w-md mx-auto px-6 py-3 border border-gray-200 border-opacity-75 rounded-lg shadow-lg w-full space-x-6 flex items-center bg-white"><button class="focus:outline-none">
        <fa-stub icon="search" class="text-gray-500"></fa-stub>
      </button> <input type="search" placeholder="Type some title..." class="w-full bg-transparent text-base focus:outline-none"> <button type="submit" class="bg-indigo-500 text-white py-2 text-sm px-3 rounded focus:outline-none">
        Search
      </button></form>
  </div>
</div>
`;
